django/db/models/sql/query.py

```diff
@@ -65,22 +65,46 @@ def get_field_names_from_opts(opts):
     )
 
 
+def get_paths_from_expression(expr):
+    if isinstance(expr, F):
+        yield expr.name
+    elif hasattr(expr, "flatten"):
+        for child in expr.flatten():
+            if isinstance(child, F):
+                yield child.name
+            elif isinstance(child, Q):
+                yield from get_children_from_q(child)
+
+
 def get_children_from_q(q):
     for child in q.children:
         if isinstance(child, Node):
             yield from get_children_from_q(child)
         else:
-            yield child
+            if isinstance(child, tuple):
+                lhs, rhs = child
+                yield lhs
+                if hasattr(rhs, "resolve_expression"):
+                    yield from get_paths_from_expression(rhs)
+            if hasattr(child, "resolve_expression"):
+                yield from get_paths_from_expression(child)
+
+
+def get_child_with_renamed_prefix(prefix, replacement, child):
+    if isinstance(child, Node):
+        return rename_prefix_from_q(prefix, replacement, child)
+    if isinstance(child, tuple):
+        lhs, rhs = child
+        lhs = lhs.replace(prefix, replacement, 1)
+        if hasattr(rhs, "resolve_expression"):
+            rhs = get_child_with_renamed_prefix(prefix, replacement, rhs)
+        return (lhs, rhs)
+    return child.copy()
```

Comment:
Pretty sure you need to handle the case where a child has a `resolve_expression` as well. You'll likely need to rely on `get_source_expressions` and `set_source_expressions`

For example, `Q(GreaterThan(F("books__author__age"), F("books__author__num_books")))` with a prefix change of `books__author -> filtered_authors` should be turned into `Q(GreaterThan(F("filtered_authors__age"), F("filtered_authors__num_books")))`.

Currently only a copy is returned.